### PR TITLE
Studio full screen / PiP playback

### PR DESCRIPTION
**Pull request by vargaat**
refs: MBL-17495
affects: Student, Teacher, Parent
release note: Embedded Studio videos now can be played in full screen or in picture-in-picture mode on iOS 17.

test plan:
- Test Studio launched from the side menu.
- Test Studio as an embedded LTI tool where rich text is available.
- Full screen mode should be available.
- PiP mode should be available after going full screen with a non-muted video.

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

<table>
<tr><th colspan="2">PiP Mode</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created
- [x] A11y checked
- [ ] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [x] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Parent Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
#### Release Note: 
Embedded Studio videos now can be played in full screen or in picture-in-picture mode on iOS 17.

#### Affected Apps: Student, Teacher, Parent

[MBL-17495](https://instructure.atlassian.net/browse/MBL-17495)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.55% | 91.54% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bd144bca00b114e7e549e9ad849f16eda9296940
</p>

**Review by rh12 — APPROVED** [review]
Code + 1
QA + 1

1. Not a big deal, but when entering/exiting full screen the animation is a bit weird. When entering full screen it oversizes the video first and then shrinks it back to fit fullscreen after. It's a fast but not so smooth animation. It is more noticeable in light backgrounds (ie in Studio side menu). Again, IMHO it's not that big of a deal, if there is no quick fix for it.
2. VoiceOver in fullscreen mode allows the user to navigate between UI elements of the screen "behind" the video.

**Comment by rh12** [comment]
One question: why do we disallow PiP when video is muted?
In case it is done because it seems not that useful: even if we don't see a trivial usecase for it, do we gain something by restricting it?

File: Core/CoreTests/LTI/View/StudioViewControllerTests.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+@testable import Core
+import XCTest
+
+class StudioViewControllerTests: XCTestCase {
+
+    func testUsesForcedDarkMode() {
```
[the remark this answers is not in the document]
Comment (vargaat) [reply Core/CoreTests/LTI/View/StudioViewControllerTests.swift:24]: True that.

File: Core/Core/Extensions/UIViewExtensions.swift [hunk 2]
```diff
@@ -126,4 +126,10 @@ extension UIView {
             pin(inside: superview, leading: nil, trailing: nil, top: 0, bottom: 0)
         }
     }
+
+    public func constraintsAffecting(view: UIView) -> [NSLayoutConstraint] {
+        constraints.filter { constraint in
+            constraint.firstItem as? NSObject == view || constraint.secondItem  as? NSObject == view
```
[the remark this answers is not in the document]
Comment (vargaat) [reply Core/Core/Extensions/UIViewExtensions.swift:132]: Fixed.

**Comment by vargaat** [comment]
> One question: why do we disallow PiP when video is muted? In case it is done because it seems not that useful: even if we don't see a trivial usecase for it, do we gain something by restricting it?

It's not that we disallow it but that's how the native video player behaves. You can test that the behavior is the same for embedded youtube videos.

**Review by rh12 — APPROVED** [review]

**Review by szabinst — APPROVED** [review]
QA+1

**Review by ndrsszsz — APPROVED** [review]
QA +1

**Pull request merged**